firestore-counter/extension.yaml

```diff
@@ -36,13 +36,28 @@ contributors:
 roles:
   - role: datastore.user
     reason: Allows the extension to aggregate Cloud Firestore counter shards.
+  - role: pubsub.publisher
+    reason: Allows the HTTPS function to trigger the PubSub handler.
 
 resources:
-  - name: controller
+  - name: controllerCore
     type: firebaseextensions.v1beta.function
     description:
       Scheduled to run every minute.
       This function either aggregates shards itself, or it schedules and monitors workers to aggregate shards.
+    properties:
+      sourceDirectory: .
+      location: ${LOCATION}
+      maxInstances: 1
+      eventTrigger:
+        eventType: google.pubsub.topic.publish
+        resource: projects/${PROJECT_ID}/topics/${EXT_INSTANCE_ID}
+
+  - name: controller
+    type: firebaseextensions.v1beta.function
+    description:
+      Maintained for backwards compatibility.
+      This function will relay the message to the onPublish function.
```

Comment:
If I understand what this function does in the grand scheme of things, I would recommend the following edit:

This function relays a message to the extension's Pub/Sub topic to trigger the controllerCore function.

Reply:
Yes,that's more accurate.

Reply:
updated.  thx!